posthog/api/dashboards/dashboard.py

```diff
@@ -489,16 +491,25 @@ def get_tiles(self, dashboard: Dashboard) -> Optional[list[ReturnDict]]:
             group_properties={"organization": {"id": str(team.organization_id)}},
         )
 
+        # Get layout size and limit parameters for progressive loading
+        limit_tiles, layout_size = self._get_progressive_loading_params()
+
         # Sort tiles by layout to ensure insights are computed in order of appearance on dashboard
-        # sm more common than xs, so we sort by sm
+        # Use the specified layout size to get the correct order for the current viewport
         sorted_tiles = sorted(
             tiles,
             key=lambda tile: (
-                tile.layouts.get("sm", {}).get("y", 100),
-                tile.layouts.get("sm", {}).get("x", 100),
+                tile.layouts.get(layout_size, {}).get("y", 100),
+                tile.layouts.get(layout_size, {}).get("x", 100),
             ),
         )
 
+        # Apply tile limit if specified, but only if there are enough tiles to make it worthwhile
+        # For dashboards with < 10 tiles, return all tiles even if limit_tiles is specified
+        total_tiles = len(sorted_tiles)
+        if limit_tiles is not None and limit_tiles > 0 and total_tiles >= 10:
```

Comment:
Why do we need/want this? We could just request a limit of 10 insights from the frontend.